flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/reader/SingleThreadMultiplexSourceReaderBase.java

```diff
@@ -113,4 +115,26 @@ public SingleThreadMultiplexSourceReaderBase(
             SourceReaderContext context) {
         super(elementsQueue, splitFetcherManager, recordEmitter, config, context);
     }
+
+    /**
+     * This constructor behaves like {@link #SingleThreadMultiplexSourceReaderBase(Supplier,
+     * RecordEmitter, Configuration, SourceReaderContext)}, but accepts a specific {@link
+     * FutureCompletingBlockingQueue}, {@link RecordEvaluator} and {@link
+     * SingleThreadFetcherManager}.
+     */
+    public SingleThreadMultiplexSourceReaderBase(
+            FutureCompletingBlockingQueue<RecordsWithSplitIds<E>> elementsQueue,
+            SingleThreadFetcherManager<E, SplitT> splitFetcherManager,
+            RecordEmitter<E, T, SplitStateT> recordEmitter,
+            @Nullable RecordEvaluator<T> eofRecordEvaluator,
```

Comment:
The changes in the kafka connector will raise in another PR after merging this PR.  This PR focuses on the module flink-connector-base.

Reply:
Not sure if we have the precedence of splitting a FLIP's implementation into multiple PRs. I am a bit concerned that it can make the FLIP's PR harder to track and manage. It should be OK to split the implementation into multiple commits in the same PR.

Since we can not get concrete user-side benefits before making the changes in the kafka connector etc, would it be simpler to put all of them in the same PR?